state-synchronizer/src/coordinator.rs

```diff
@@ -384,6 +404,36 @@ impl<T: ExecutorProxyTrait> SyncCoordinator<T> {
         Ok(())
     }
 
+    /// The choice of the LedgerInfo in the response follows the following logic:
+    /// * response LI is either the requested target or the highest local LI if target is None.
+    /// * if the response LI would not belong to `request_epoch`, change
+    /// the response LI to the LI that is terminating `request_epoch`.
+    fn choose_response_li(
+        &self,
+        known_version: u64,
+        request_epoch: u64,
+        target: Option<LedgerInfoWithSignatures>,
+    ) -> Result<LedgerInfoWithSignatures> {
+        let mut target_li = target.unwrap_or_else(|| self.local_state.highest_local_li.clone());
+        if target_li.ledger_info().epoch() > request_epoch {
+            let end_of_epoch_li = self
+                .executor_proxy
+                .get_epoch_proof(request_epoch)?
```

Comment:
just fyi this function will return error if `request_epoch` is more than 100 epoch away, @wqfish or @msmouse will figure out a better way

Reply:
Yea, absolutely! We probably need an iterative solution similar to what state synchronizer does today for chunk limits: given a known epoch return up to X validator changes in a message. In this specific case this is trivial because we just need the next epoch info, but for consensus it might be slightly more interesting. Opened task #1829 for tracking it.